packages/shared/src/components/alert/PointedAlert.tsx

```diff
@@ -0,0 +1,134 @@
+import React, { CSSProperties, ReactElement, ReactNode, useRef } from 'react';
+import classNames from 'classnames';
+import XIcon from '../icons/Close';
+import { Button } from '../buttons/Button';
+import {
+  PointedAlertMessage,
+  PointedAlertCopy,
+  PointedAlertWrapper,
+  PointedAlertContainer,
+} from './common';
+import Pointer, { PointerColor } from './Pointer';
+
+interface ClassName {
+  container?: string;
+  pointer?: string;
+  wrapper?: string;
+  message?: string;
+  label?: string;
+}
+
+type OffsetXY = [number?, number?];
+
+export enum AlertPlacement {
+  Top = 'top',
+  Right = 'right',
+  Bottom = 'bottom',
+  Left = 'left',
+}
+
+const messageBorder: Record<PointerColor, string> = {
+  [PointerColor.Cabbage]: 'border-theme-status-cabbage',
+  [PointerColor.Success]: 'border-theme-status-success',
+};
+
+const pointerClasses = {
+  [AlertPlacement.Top]: 'rotate-180',
+  [AlertPlacement.Right]: '-rotate-90',
+  [AlertPlacement.Left]: 'rotate-90',
+};
+
+const alertContainerClasses: Record<AlertPlacement, string> = {
+  top: '-translate-y-full top-0 flex-col-reverse',
+  right: 'translate-x-full right-0 flex-row',
+  left: '-translate-x-full left-0 flex-row-reverse',
+  bottom: 'translate-y-full bottom-0 flex-col',
+};
+
+const getContainerStyle = (
+  [xOffset = 0, yOffset = 0]: OffsetXY = [0, 0],
+): Record<AlertPlacement, CSSProperties> => ({
+  left: { left: xOffset },
+  right: { right: xOffset },
+  bottom: { bottom: yOffset },
+  top: { top: yOffset },
+});
+
+export interface PointedAlertProps {
+  className?: ClassName;
+  children: ReactNode;
+  message: ReactNode;
+  placement?: AlertPlacement;
+  offset?: OffsetXY;
+  color?: PointerColor;
+  isAlertDisabled?: boolean;
+  onClose?: () => unknown;
+}
+
+const DEFAULT_POINTER_LENGTH = 6;
+const getMessageStyle = (
+  length = DEFAULT_POINTER_LENGTH,
+): Record<AlertPlacement, CSSProperties> => ({
+  left: { marginRight: length },
+  right: { marginLeft: length },
+  bottom: { marginBottom: -length },
+  top: { marginTop: -length },
+});
+
+export default function PointedAlert({
+  placement = AlertPlacement.Right,
+  color = PointerColor.Cabbage,
+  className = {},
+  children,
+  message,
+  offset,
+  isAlertDisabled,
+  onClose,
+}: PointedAlertProps): ReactElement {
+  const pointerRef = useRef<HTMLDivElement>();
+  const rect = pointerRef?.current?.getBoundingClientRect?.();
+
+  if (isAlertDisabled) {
+    return <>{children}</>;
+  }
```

Comment:
In which scenario do we use this version?

Reply:
When we now hide the alert and just show the children we wrapped.